File: Random/1.Brick Wall.py
#    Author : Mohamed Yousef 
#    Date   : 2022-12-18
import sys,math,bisect,collections,itertools,heapq
from collections import defaultdict,deque
sys.setrecursionlimit(50000) #To increase Recursion depth in py
def solve(wall):  
    myDict=defaultdict(int)
    n=len(wall)
    for i in range(n):
        for j in range(len(wall[i])):
            if j>0:
                wall[i][j]+=wall[i][j-1]
            myDict[wall[i][j]]+=1
    ans=0
    for key,value in myDict.items():
        if wall[0][-1]!=key:
            ans=max(value,ans)
    return n-ans
print(solve([[2],[2],[2]]))

File: Random/1.Brick Wall.js
/*
    Author : Mohamed Yousef 
    Date   : 2022-12-20
*/
let solve=function(wall){
    let defaultDict=new Proxy({},{
        get:(target,Name)=> Name in target?target[Name]:0
    });
    let ans=0
    n=wall.length;
    for(let i=0;i<n;i++){
        for(let j=0;j<wall[i].length;j++){
            if (j>0)
                wall[i][j]+=wall[i][j-1]
            defaultDict[wall[i][j]]+=1
        
    }
    }
    // console.log(wall[0].at(-1));
    for (let [key,value] of Object.entries(defaultDict)){
        if(key!=wall[0].at(-1)){
            ans=Math.max(value,ans)
        }
    }
    return n-ans
}
console.log(solve([[1,2,2,1],[3,1,2],[1,3,2],[2,4],[3,1,2],[1,3,1,1]]));